packages/react-instantsearch-core/src/core/__tests__/createInstantSearchManager.js

```diff
@@ -264,6 +264,51 @@ describe('createInstantSearchManager', () => {
 
       expect(Object.keys(searchClient.cache)).toHaveLength(0);
     });
+
+    it('when using algoliasearch@v4, it overrides search only once', () => {
+      const searchClient = algoliasearch('appId', 'apiKey', {
+        _cache: true,
+      });
+
+      // Skip this test with Algoliasearch API Client < v4, as
+      // search does not need to be overridden.
+      if (!searchClient.transporter) {
+        return;
+      }
+
+      const resultsState = {
+        rawResults: [
+          {
+            index: 'indexName',
+            query: 'query',
+          },
+        ],
+        state: {
+          index: 'indexName',
+          query: 'query',
+        },
+      };
+
+      const originalSearch = algoliasearch.search;
+
+      createInstantSearchManager({
+        indexName: 'index',
+        searchClient,
+        resultsState,
+      });
+
+      expect(searchClient.search).not.toBe(originalSearch);
+
+      searchClient.search = 'already-overridden';
+
+      createInstantSearchManager({
+        indexName: 'index',
+        searchClient,
+        resultsState,
+      });
+
+      expect(searchClient.search).toEqual('already-overridden');
```

Comment:
Since this is a string, I believe this test doesn't actually test the new behavior. We should attach an actual function to check that the reference changes only once, right?

Reply:
what we're checking is that after the search client has been "hydrated" once, it will no longer change, so if we manually change it to something we can easily check the reference of (a string), it will no longer change. Before this PR it would have changed a second time

Reply:
Gotcha! We don't make sure that reference says the same though, which is what our code does.

@nunomaduro It would make more sense to attach `jest.fn()` to `.search` and to check the function reference. Wdyt?

Reply:
Good idea.

Addressed here: https://github.com/algolia/react-instantsearch/pull/2926/commits/4c28cb98bc88f8ad1e5b2f2bb23734e25961a0f2.